src/ZkFold/Symbolic/Compiler.hs

```diff
@@ -30,18 +31,33 @@ import           ZkFold.Symbolic.Compiler.Arithmetizable
     8. ZkFold.Symbolic.Compiler
 -}
 
--- | Arithmetizes an argument by feeding an appropriate amount of inputs.
-solder :: forall a f . Arithmetizable a f => f -> [ArithmeticCircuit a]
-solder f = arithmetize f $ circuits $ replicateA (inputSize @a @f) input
+-- | Precompiles function `f` into a sequence of arithmetic circuits.
+precompile
+  :: forall a f . (Arithmetic a, FunctionSpace (ArithmeticCircuit a) f)
```

Comment:
How will `FunctionalSpace (ArithmeticCircuit a) f` will translate into ArithmeticCircuits with multiple outputs? It won't be a sequence of circuits anymore, just one circuit with several inputs

Reply:
Once we have both structured inputs and structured multi-outputs, then the compile function will look something like this:

```Haskell
compile
  :: (Arithmetic a, SymbolicFunction (ArithmeticCircuit i Par1 a) f)
  => f -> ArithmeticCircuit (InputSpace f) (OutputSpace f) a
compile f = optimize (solder (curryV f (desolder idCircuit)))
```

Reply:
```Haskell
idCircuit :: SymbolicData i => Circuit i i a
idCircuit = mempty {acOutputs = iterateV 1 succ}
```